### PR TITLE
[v1] Add support for Time and Regexp

### PR DESCRIPTION
**Pull request by spraints**
This is #19, ported to the v1-release branch.

cc @brianmario @vmg @kivikakk @carlosmn @piki

### CODE REVIEW

**Comment by spraints** [comment]
@vmg when you have a moment, can you do a code or security review of this PR, too? Its intent is the same as #19, but its implementation is slightly different because of the differences between v1 and v2. This is what we'll actually end up running in prod, since v1 is what we already have installed.

File: ext/mochilo/mochilo_pack.c [hunk 1]
```diff
@@ -73,6 +73,56 @@ void mochilo_pack_bignum(mochilo_buf *buf, VALUE rb_bignum)
 	}
 }
 
+#ifdef HAVE_RUBY_ENCODING_H
```
Comment (brianmario) [inline ext/mochilo/mochilo_pack.c]: This could be saved for another pull request for sure but - I'm pretty sure we dropped support for anything older than Ruby 1.9 so we probably don't need these checks anymore.
